src/main/resources/guillotine/resolver.js

```diff
@@ -29,3 +42,35 @@ exports.MenuItem_content_Resolver = function (env) {
         });
     });
 };
+
+exports.MenuItem_path_Resolver = function (env) {
+    const basePath = env.localContext._menuAppBasePath;
+    return basePath === '/' ? env.source.path : env.source.path.replace(basePath, '');
+};
+
+function resolveMenuSource(env) {
+    return ctxLib.executeInContext(env, function () {
+        const key = env.localContext.siteKey;
+
+        if (!key || key.trim() === '') {
```

Comment:
to be sure that it is not empty string

Reply:
OK, it is sort of optimization, I guess, to avoid calling getSite, but IMO, it is getSite which should have such optimization, not the code around it.